src/components/NewCommentForm.tsx

```
@@ -1,8 +1,168 @@
+    setTouched({ name: true, email: true, body: true });
+
+    return !Object.values(newErrors).some(error => error);
+  }, [formData.name, formData.email, formData.body, validateField]);
+
+  const handleReset = useCallback(() => {
+    setFormData({ name: '', email: '', body: '' });
+    setErrors({ name: false, email: false, body: false });
+    setTouched({ name: false, email: false, body: false });
+    setIsSubmitting(false);
+    setSubmitError(null);
+  }, []);
+
+  const clearFormAfterSubmit = useCallback(() => {
+    setFormData(prev => ({ ...prev, body: '' }));
+    setErrors({ name: false, email: false, body: false });
+    setTouched(prev => ({ ...prev, body: false }));
+    setSubmitError(null);
+  }, []);
+
+  const handleSubmit = useCallback(async () => {
+    if (!validateForm() || !onAddComment || isSubmitting) {
+      return;
+    }
+
+    setIsSubmitting(true);
+    setSubmitError(null);
+
+    try {
+      const commentData = {
+        name: formData.name,
+        email: formData.email,
+        body: formData.body,
+      };
+
+      await Promise.resolve(onAddComment(commentData));
+      clearFormAfterSubmit();
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Failed to add comment';
+
+      setSubmitError(errorMessage);
+      // Re-throw to allow parent component to handle the error
+      throw error;
+    } finally {
+      setIsSubmitting(false);
+    }
+  }, [
+    formData.name,
+    formData.email,
+    formData.body,
+    onAddComment,
+    isSubmitting,
+    validateForm,
+    clearFormAfterSubmit,
+  ]);
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form
+      data-cy="NewCommentForm"
+      onReset={handleReset}
+      onSubmit={async e => {
+        e.preventDefault();
+        try {
+          await handleSubmit();
+        } catch (error) {
+          // Error is already handled in handleSubmit
+        }
```

The inline onSubmit handler catches errors from handleSubmit and ignores them (`// Error is already handled in handleSubmit`). Because handleSubmit re-throws (see lines 138-139), that rethrow is caught and discarded here. Decide where errors should be handled. If you want parent-level handling/retries, remove this try/catch so the error can bubble; if you handle it here, remove the rethrow in handleSubmit.

This is the catch that swallows the thrown error from handleSubmit.